src/containerapp/azext_containerapp/custom.py

```diff
@@ -0,0 +1,954 @@
+    yaml_containerapp = process_loaded_yaml(load_yaml_file(file_name))
+    if type(yaml_containerapp) != dict:
+        raise ValidationError('Invalid YAML provided. Please see https://docs.microsoft.com/azure/container-apps/azure-resource-manager-api-spec#examples for a valid containerapps YAML spec.')
+
+    if not yaml_containerapp.get('name'):
+        yaml_containerapp['name'] = name
+    elif yaml_containerapp.get('name').lower() != name.lower():
+        logger.warning('The app name provided in the --yaml file "{}" does not match the one provided in the --name flag "{}". The one provided in the --yaml file will be used.'.format(
+            yaml_containerapp.get('name'), name))
+    name = yaml_containerapp.get('name')
+
+    if not yaml_containerapp.get('type'):
+        yaml_containerapp['type'] = 'Microsoft.App/containerApps'
+    elif yaml_containerapp.get('type').lower() != "microsoft.app/containerapps":
+        raise ValidationError('Containerapp type must be \"Microsoft.App/ContainerApps\"')
+
+    current_containerapp_def = None
+    containerapp_def = None
+    try:
+        current_containerapp_def = ContainerAppClient.show(cmd=cmd, resource_group_name=resource_group_name, name=name)
+    except Exception as ex:
+        pass
+
+    if not current_containerapp_def:
+        raise ValidationError("The containerapp '{}' does not exist".format(name))
+
+    # Deserialize the yaml into a ContainerApp object. Need this since we're not using SDK
+    try:
+        deserializer = create_deserializer()
+
+        containerapp_def = deserializer('ContainerApp', yaml_containerapp)
+    except DeserializationError as ex:
+        raise ValidationError('Invalid YAML provided. Please see https://docs.microsoft.com/azure/container-apps/azure-resource-manager-api-spec#examples for a valid containerapps YAML spec.')
+
+    # Remove tags before converting from snake case to camel case, then re-add tags. We don't want to change the case of the tags. Need this since we're not using SDK
+    tags = None
+    if yaml_containerapp.get('tags'):
+        tags = yaml_containerapp.get('tags')
+        del yaml_containerapp['tags']
+
+    containerapp_def = _convert_object_from_snake_to_camel_case(_object_to_dict(containerapp_def))
+    containerapp_def['tags'] = tags
+
+    # After deserializing, some properties may need to be moved under the "properties" attribute. Need this since we're not using SDK
+    containerapp_def = process_loaded_yaml(containerapp_def)
+
+    _get_existing_secrets(cmd, resource_group_name, name, current_containerapp_def)
+
+    update_nested_dictionary(current_containerapp_def, containerapp_def)
+
+    # Remove "additionalProperties" and read-only attributes that are introduced in the deserialization. Need this since we're not using SDK
+    _remove_additional_attributes(current_containerapp_def)
+    _remove_readonly_attributes(current_containerapp_def)
+
+    '''
```

Comment:
Is this comment block needed? Are there any open questions here that need to be addressed?

Reply:
Removed